custom_components/smartir/light.py

```diff
@@ -231,26 +245,40 @@ async def async_turn_on(self, **params):
             target = params.get(ATTR_COLOR_TEMP_KELVIN)
             old_color_temp = DeviceData.closest_match(self._colortemp, self._colortemps)
             new_color_temp = DeviceData.closest_match(target, self._colortemps)
-            _LOGGER.debug(
-                f"Changing color temp from {self._colortemp}K step {old_color_temp} to {target}K step {new_color_temp}"
-            )
-
-            steps = new_color_temp - old_color_temp
-            did_something = True
-            if steps < 0:
-                cmd = CMD_COLORMODE_WARMER
-                steps = abs(steps)
-            else:
-                cmd = CMD_COLORMODE_COLDER
-
-            if steps > 0 and cmd:
-                # If we are heading for the highest or lowest value,
-                # take the opportunity to resync by issuing enough
-                # commands to go the full range.
-                if new_color_temp == len(self._colortemps) - 1 or new_color_temp == 0:
-                    steps = len(self._colortemps)
+            final_color_temp = f"{self._colortemps[new_color_temp]}"
+            if (
+                CMD_COLORTEMPERATURE in self._commands
+                and isinstance(self._commands[CMD_COLORTEMPERATURE], dict)
+                and final_color_temp in self._commands[CMD_COLORTEMPERATURE]
+            ):
+                _LOGGER.debug(
+                    f"Changing color temp from {self._colortemp}K to {target}K using found remote command for {final_color_temp}K"
+                )
+                did_something = True
+                found_command = self._commands[CMD_COLORTEMPERATURE][final_color_temp]
                 self._colortemp = self._colortemps[new_color_temp]
-                await self.send_command(cmd, steps)
+                await self.send_remote_command(found_command)
+            else:
+                _LOGGER.debug(
+                    f"Changing color temp from {self._colortemp}K step {old_color_temp} to {target}K step {new_color_temp}"
+                )
+
+                steps = new_color_temp - old_color_temp
+                if steps < 0:
+                    cmd = CMD_COLORMODE_WARMER
+                    steps = abs(steps)
+                else:
+                    cmd = CMD_COLORMODE_COLDER
+
+                if steps > 0 and cmd:
+                    # If we are heading for the highest or lowest value,
+                    # take the opportunity to resync by issuing enough
+                    # commands to go the full range.
+                    if new_color_temp == len(self._colortemps) - 1 or new_color_temp == 0:
+                        steps = len(self._colortemps)
+                    did_something = True
```

Comment:
This line (279) needs to move back up to above line 267, to avoid sending "on" commands when a color temp is given but does not result in a change.

Reply:
Moved the assignation to line 245, affecting the both branches of the if.